common/flatpak-run.c

```diff
@@ -3165,6 +3169,88 @@ regenerate_ld_cache (GPtrArray    *base_argv_array,
   return glnx_steal_fd (&ld_so_fd);
 }
 
+/* Check that this user is actually allowed to run this app. When running
+ * from the gnome-initial-setup session, an app filter might not be available. */
+static gboolean
+check_parental_controls (const char     *app_ref,
+                         FlatpakDeploy  *deploy,
+                         GCancellable   *cancellable,
+                         GError        **error)
+{
+#ifdef HAVE_LIBMALCONTENT
+  g_auto(GStrv) app_ref_parts = NULL;
+  g_autoptr(MctManager) manager = NULL;
+  g_autoptr(MctAppFilter) app_filter = NULL;
+  g_autoptr(GAsyncResult) app_filter_result = NULL;
+  g_autoptr(GDBusConnection) system_bus = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  app_ref_parts = flatpak_decompose_ref (app_ref, error);
+  if (app_ref_parts == NULL)
+    return FALSE;
+
+  system_bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
+  if (system_bus == NULL)
+    return FALSE;
```

Comment:
I wonder if there is some easy check we could do that we actually got the real system bus here. Maybe we can verify that the peer uid is 0? That kills an easy route around this.

Reply:
Interesting idea. It might be possible to do that by extracting the `GSocket` from the `GSocketConnection` returned by `g_dbus_connection_get_stream()`, then calling `g_socket_get_credentials()` on it. That *should* give the credentials of the system D-Bus daemon process.

However, it’s quite a faff and would make testing significantly harder unless the code which is tested differs from that in production. I’ll see if I can fit something sensible in.

Reply:
Filed https://github.com/flatpak/flatpak/issues/2994 as a follow-up for this.